File: visa-dev-samplecode-master/vdp-ruby/test/vta/manage_notifications_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class ManageNotificationsTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @visa_api_client = VisaAPIClient.new
    @notificationSubscriptionRequest = '''{
      "contactType": "''' + @config['vtaNotificationContactType'] + '''",
      "contactValue": "john@visa.com",
      "emailFormat": "None",
      "last4": "''' + @config['vtaCreateCustomerLastFour'] + '''",
      "phoneCountryCode": "en-us",
      "platform": "None",
      "preferredLanguageCode": "''' + @config['vtaPreferredLanguageCode'] + '''",
      "serviceOffering": "WelcomeMessage",
      "serviceOfferingSubType": "WelcomeMessage",
      "substitutions": {}
    }'''
  end
  
  def test_notificationSubscription
    base_uri = 'vta/'
    resource_path = "v3/communities/#{@config['vtaCommunityCode']}/portfolios/#{@config['vtaPortfolioNumber']}/customers/#{@config['vtaCustomerId']}/notifications"
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Notification Subscriptions Test", "post", @notificationSubscriptionRequest, {'ServiceId' => @config['vtaServiceId']})
    assert_equal("201", response_code, "Notification Subscriptions Test failed")
  end
end

File: visa-dev-samplecode-master/vdp-node/test/vta/managenotificationstest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Manage Notifications', function() {
	var visaAPIClient = new VisaAPIClient();
	var notificationSubscriptionRequest = JSON.stringify({
        "contactType": config.vtaNotificationContactType,
        "contactValue": "john@visa.com",
        "emailFormat": "None",
        "last4": config.vtaCreateCustomerLastFour,
        "phoneCountryCode": "en-us",
        "platform": "None",
        "preferredLanguageCode": config.vtaPreferredLanguageCode,
        "serviceOffering": "WelcomeMessage",
        "serviceOfferingSubType": "WelcomeMessage",
        "substitutions": {}
     });

	it('Notification Subscriptions Test',function(done) {
		this.timeout(10000);
		var baseUri = 'vta/';
		var resourcePath = 'v3/communities/'+ config.vtaCommunityCode +'/portfolios/' + config.vtaPortfolioNumber +'/customers/' + config.vtaCustomerId+ '/notifications';
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, notificationSubscriptionRequest, 'POST', {'ServiceId' : config.vtaServiceId}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 201);
			} else {
				assert(false);
			}
			done();
		});
	});
});
